syntax = "proto3";
import "ipss.proto";

message Data {
    float score = 1;
    int32 index = 2;
}

message SolverWithConfig {
    ipss.Service definition = 1;
    map<string, ipss.Any> enviroment_variables = 2;
}

message DataSetInstance {
    SolverWithConfig solver = 1;
    map<string, Data> data = 2;
    string hash = 3; // identifica el Solver con una configuracion determinada.
}

message DataSet {
    repeated DataSetInstance data = 1;
}
